### PR TITLE
fix(API security): Don't allow viewing "private" attendees if user can't

### PR DESCRIPTION
**Pull request by Camwyn**
… edit private posts.

Add a filter to the Repository (done in tribe-common) and utilize it when making API calls - if the
user cannot edit private posts then they can only view attendees from event that have the "Show
attendees list on event page" checkbox checked.

Requires: https://github.com/moderntribe/tribe-common/pull/1222 (included in PR)

:ticket: https://central.tri.be/issues/133333

📷 http://p.tri.be/8hslf3 (as admin, 72 is "private")
📸 http://p.tri.be/gWpFmE (as non-admin)

### CODE REVIEW

File: src/Tribe/Attendee_Repository.php [hunk 1]
```diff
@@ -395,6 +396,23 @@ public function filter_by_event_status( $event_status ) {
...
 	}
 
+	/**
+	 * Filters attendee to only get those related to posts with "Show attendees list on event page" set to true.
+	 *
+	 *
+	 * @since TBD
+	 */
+	public function filter_by_show_attendees() {
+		$can_read_private_posts = current_user_can( 'read_private_posts' );
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Repository.php]: This var is setup but not used below

**Comment by Camwyn** [comment]
@sc0ttkclark Yeah, so those REST tests failed big-time... 😢

**Comment by sc0ttkclark** [comment]
I have one concern about this I'd like to talk further about

**Comment by sc0ttkclark** [comment]
As per our discussion, we should update this code to check for hide = 1 OR hide IS NULL (for cases where the hide is not set at all)

**Pull request merged**